pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,245 @@
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+
+	kanister "github.com/kanisterio/kanister/pkg"
+	"github.com/kanisterio/kanister/pkg/consts"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/format"
+	kankopia "github.com/kanisterio/kanister/pkg/kopia"
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	kerrors "github.com/kanisterio/kanister/pkg/kopia/errors"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/param"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	BackupDataUsingKopiaServerFuncName = "BackupDataUsingKopiaServer"
+	// BackupDataTagsKeyArg is the key used for returning snapshot tags
+	BackupDataTagsKeyArg = "snapshotTags"
+)
+
+type backupDataUsingKopiaServerFunc struct{}
+
+func init() {
+	err := kanister.Register(&backupDataUsingKopiaServerFunc{})
+	if err != nil {
+		return
+	}
+}
+
+var _ kanister.Func = (*backupDataUsingKopiaServerFunc)(nil)
+
+func (*backupDataUsingKopiaServerFunc) Name() string {
+	return BackupDataUsingKopiaServerFuncName
+}
+
+func (*backupDataUsingKopiaServerFunc) RequiredArgs() []string {
+	return []string{
+		BackupDataContainerArg,
+		BackupDataIncludePathArg,
+		BackupDataNamespaceArg,
+		BackupDataPodArg,
+		kankopia.KopiaUserPassphraseArg,
+		kankopia.KopiaTLSCertSecretDataArg,
+	}
+}
+
+func (*backupDataUsingKopiaServerFunc) Arguments() []string {
+	return []string{
+		BackupDataContainerArg,
+		BackupDataIncludePathArg,
+		BackupDataNamespaceArg,
+		BackupDataPodArg,
+		kankopia.KopiaUserPassphraseArg,
+		kankopia.KopiaTLSCertSecretDataArg,
+		BackupDataTagsKeyArg,
+	}
+}
+
+func (*backupDataUsingKopiaServerFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]any) (map[string]any, error) {
+	//TODO implement me
+	var (
+		container      string
+		err            error
+		includePath    string
+		namespace      string
+		pod            string
+		userPassphrase string
+		cert           string
+		tagsStr        string
+	)
+	if err = Arg(args, BackupDataContainerArg, &container); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataIncludePathArg, &includePath); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataNamespaceArg, &namespace); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataPodArg, &pod); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, kankopia.KopiaUserPassphraseArg, &userPassphrase); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, kankopia.KopiaTLSCertSecretDataArg, &cert); err != nil {
+		return nil, err
+	}
+	if err = OptArg(args, BackupDataTagsKeyArg, &tagsStr, ""); err != nil {
+		return nil, err
+	}
+
+	var tags []string = nil
+	if tagsStr != "" {
+		tags = strings.Split(tagsStr, ",")
+	}
+
+	fingerprint, err := kankopia.ExtractFingerprintFromCertificateJSON(cert)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch Kopia API Server Certificate Secret Data from blueprint")
```

Comment:
This comment seems incorrect to me. We are not fetching the secret data from blueprint right?

Reply:
Sure ! will make the changes